sdk/tables/Azure.Data.Tables/tests/perf/Infrastructure/TablesPerfTest.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Data.Tables.Tests;
+using Azure.Test.Perf;
+
+namespace Azure.Data.Tables.Performance
+{
+    public abstract class TablesPerfTest : PerfTest<TablePerfOptions>
+    {
+        private const string stringValue = "This is a string";
+        private Guid guid = Guid.NewGuid();
+        private DateTime dt = DateTime.UtcNow;
+        private byte[] binary = new byte[] { 0x01, 0x02, 0x03, 0x04, 0x05 };
+        private static TablesTestEnvironment _environment = new TablesTestEnvironment();
+        private const string PartitionKey = "performance";
+        private const string TableName = "perfTestTable";
+
+        protected TablesTestEnvironment TestEnvironment => _environment;
+
+        protected TableTransactionalBatch GetBatch() => Client?.CreateTransactionalBatch(PartitionKey);
+
+        protected TableClient Client;
+
+        public TablesPerfTest(TablePerfOptions options) : base(options)
+        { }
+
+        public override async Task GlobalSetupAsync()
+        {
+            var serviceUri = Options.EndpointType switch
+            {
+                TableEndpointType.Storage => TestEnvironment.StorageUri,
+                TableEndpointType.CosmosTable => TestEnvironment.CosmosUri,
+                _ => throw new NotSupportedException("Unknown endpoint type")
+            };
+
+            var accountName = Options.EndpointType switch
+            {
+                TableEndpointType.Storage => TestEnvironment.StorageAccountName,
+                TableEndpointType.CosmosTable => TestEnvironment.CosmosAccountName,
+                _ => throw new NotSupportedException("Unknown endpoint type")
+            };
+
+            var accountKey = Options.EndpointType switch
+            {
+                TableEndpointType.Storage => TestEnvironment.PrimaryStorageAccountKey,
+                TableEndpointType.CosmosTable => TestEnvironment.PrimaryCosmosAccountKey,
+                _ => throw new NotSupportedException("Unknown endpoint type")
+            };
+
+            Client = new TableClient(
+                 new Uri(serviceUri),
+                 TableName,
+                 new TableSharedKeyCredential(accountName, accountKey),
+                 new TableClientOptions());
+
+            await Client.CreateIfNotExistsAsync().ConfigureAwait(false);
+
+            await base.GlobalSetupAsync().ConfigureAwait(false);
+        }
+
+        public override async Task GlobalCleanupAsync()
+        {
+            await Client.DeleteAsync().ConfigureAwait(false);
+
+            await base.GlobalCleanupAsync().ConfigureAwait(false);
+        }
+
+        protected IEnumerable<T> GenerateEntities<T>() where T : class, ITableEntity, new()
```

Comment:
I may be overlooking something, but this seems redundant to the `EntityGenerator`... no?

Reply:
Correct - this is what is used currently.